recipes/azure-storage-cpp/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches
+import os
+
+required_conan_version = ">=1.35.0"
+
+class AzureStorageCppConan(ConanFile):
+    name = "azure-storage-cpp"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Azure/azure-storage-cpp"
+    description = "Microsoft Azure Storage Client Library for C++"
+    topics = ("azure", "cpp", "cross-platform", "microsoft", "cloud")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    short_paths = True
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compiler_version(self):
+        return {
+            "gcc": "5",
+            "Visual Studio": "14",
+            "clang": "3.4",
+            "apple-clang": "5.1",
+        }
+
+    def requirements(self):
+        self.requires("cpprestsdk/2.10.18")
+        if self.settings.os != "Windows":
+            self.requires("libxml2/2.9.10")
+            self.requires("libuuid/1.0.3")
```

Comment:
```suggestion
        self.requires("cpprestsdk/2.10.18")
        if self.settings.os != "Windows":
            self.requires("boost/1.76.0")
            self.requires("libxml2/2.9.10")
            self.requires("libuuid/1.0.3")
```

Boot is a direct dependency for non-Windows: https://github.com/Azure/azure-storage-cpp/blob/5b1c1597e6a637bc205d91342054814c8d901777/Microsoft.WindowsAzure.Storage/CMakeLists.txt#L21-L22

Don't rely on transitive dependencies coming from cpprestsdk.